### PR TITLE
Added initial solution creation with projects.

### PR DESCRIPTION
**Pull request by Khyproxios**
The initial solution was created along with the following projects:
- RestApi project to provide endpoints
- Domain for business logic
- DataAccess to handle the logic that reaches out to other services
- UnitTests to hold the tests of all the projects

### CODE REVIEW

File: Fakebook.Profile/Fakebook.Profile.DataAccess/Fakebook.Profile.DataAccess.csproj [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
```
Comment (escalonn) [inline Fakebook.Profile/Fakebook.Profile.DataAccess/Fakebook.Profile.DataAccess.csproj:4]: let's target `net5.0` for all of our projects

File: Fakebook.Profile/Fakebook.Profile.Domain/Fakebook.Profile.Domain.csproj [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
```
Comment (escalonn) [inline Fakebook.Profile/Fakebook.Profile.Domain/Fakebook.Profile.Domain.csproj:4]: see above

**Review by escalonn — CHANGES_REQUESTED** [review]

File: Fakebook.Profile/Fakebook.Profile.DataAccess/Fakebook.Profile.DataAccess.csproj (hunk 1 above)
Comment (Khyproxios) [reply Fakebook.Profile/Fakebook.Profile.DataAccess/Fakebook.Profile.DataAccess.csproj]: The project target version was changed to ```net5.0```.

File: Fakebook.Profile/Fakebook.Profile.Domain/Fakebook.Profile.Domain.csproj (hunk 2 above)
Comment (Khyproxios) [reply Fakebook.Profile/Fakebook.Profile.Domain/Fakebook.Profile.Domain.csproj]: The project target version was changed to ```net5.0```.

File: Fakebook.Profile/Fakebook.Profile.DataAccess/Fakebook.Profile.DataAccess.csproj (hunk 1 above)
Comment (escalonn) [reply Fakebook.Profile/Fakebook.Profile.DataAccess/Fakebook.Profile.DataAccess.csproj]: when you push changes addressing a comment like that, you can click the 'resolve conversation' button underneath the comment

File: Fakebook.Profile/Fakebook.Profile.RestApi/Program.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace Fakebook.ProfileRestApi
```
Comment (escalonn) [inline Fakebook.Profile/Fakebook.Profile.RestApi/Program.cs:11]: the namespace here & in startup doesnt match the folder heirarchy, which we usually expect

**Review by escalonn — CHANGES_REQUESTED** [review]

**Review by escalonn — APPROVED** [review]

**Review by escalonn — APPROVED** [review]
keep in mind that it's awkward if you want to ever name a class the same as a namespace fragment (`Profile`). c# gets a little clumsy with that, try it out and you'll see what i mean. workarounds include changing the namespace structure and not having a class with that name.

**Pull request merged**